src/listener/to_listener.rs

```diff
@@ -0,0 +1,401 @@
+                            format!("unable to parse listener from `{}`", self),
+                        )
+                    })
+                    .and_then(|url| ToListener::<State>::to_listener(url))
+            })
+    }
+}
+
+#[cfg(unix)]
+impl<State: Send + Sync + 'static> ToListener<State> for async_std::path::PathBuf {
+    type Listener = UnixListener;
+    fn to_listener(self) -> io::Result<Self::Listener> {
+        Ok(UnixListener::from_path(self))
+    }
+}
+
+#[cfg(unix)]
+impl<State: Send + Sync + 'static> ToListener<State> for std::path::PathBuf {
+    type Listener = UnixListener;
+    fn to_listener(self) -> io::Result<Self::Listener> {
+        Ok(UnixListener::from_path(self))
+    }
+}
+
+impl<State: Send + Sync + 'static> ToListener<State> for async_std::net::TcpListener {
+    type Listener = TcpListener;
+    fn to_listener(self) -> io::Result<Self::Listener> {
+        Ok(TcpListener::from_listener(self))
+    }
+}
+
+impl<State: Send + Sync + 'static> ToListener<State> for std::net::TcpListener {
+    type Listener = TcpListener;
+    fn to_listener(self) -> io::Result<Self::Listener> {
+        Ok(TcpListener::from_listener(self))
+    }
+}
+
+impl<State: Send + Sync + 'static> ToListener<State> for (&str, u16) {
+    type Listener = TcpListener;
+
+    fn to_listener(self) -> io::Result<Self::Listener> {
+        Ok(TcpListener::from_addrs(self.to_socket_addrs()?.collect()))
+    }
+}
+
+#[cfg(unix)]
+impl<State: Send + Sync + 'static> ToListener<State> for async_std::os::unix::net::UnixListener {
+    type Listener = UnixListener;
+    fn to_listener(self) -> io::Result<Self::Listener> {
+        Ok(UnixListener::from_listener(self))
+    }
+}
+
+#[cfg(unix)]
+impl<State: Send + Sync + 'static> ToListener<State> for std::os::unix::net::UnixListener {
+    type Listener = UnixListener;
+    fn to_listener(self) -> io::Result<Self::Listener> {
+        Ok(UnixListener::from_listener(self))
+    }
+}
+
+impl<State: Send + Sync + 'static> ToListener<State> for TcpListener {
+    type Listener = Self;
+    fn to_listener(self) -> io::Result<Self::Listener> {
+        Ok(self)
+    }
+}
+
+#[cfg(unix)]
+impl<State: Send + Sync + 'static> ToListener<State> for UnixListener {
+    type Listener = Self;
+    fn to_listener(self) -> io::Result<Self::Listener> {
+        Ok(self)
+    }
+}
+
+impl<State: Send + Sync + 'static> ToListener<State> for MultiListener<State> {
+    type Listener = Self;
+    fn to_listener(self) -> io::Result<Self::Listener> {
+        Ok(self)
+    }
+}
+
+impl<State: Send + Sync + 'static> ToListener<State> for ParsedListener {
+    type Listener = Self;
+    fn to_listener(self) -> io::Result<Self::Listener> {
+        Ok(self)
+    }
+}
+
+impl<State: Send + Sync + 'static> ToListener<State> for std::net::SocketAddr {
+    type Listener = TcpListener;
+    fn to_listener(self) -> io::Result<Self::Listener> {
+        Ok(TcpListener::from_addrs(vec![self]))
```

Comment:
Should `TcpListener` have `from_addr` for singles?

Reply:
It could, but what problem does that solve? I think this is the only place it's constructed with just one SocketAddr